### PR TITLE
build: add braft format patch

### PR DESCRIPTION
**Pull request by wu-hanqing**
### What problem does this PR solve?

Issue Number: #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

**Comment by Cyber-SiKu** [comment]
@wu-hanqing continue？

**Comment by wu-hanqing** [comment]
cicheck

**Comment by YunhuiChen** [comment]
cicheck

File: src/client/utils.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+#include <cstdint>
+
+namespace curve {
+namespace client {
+
+// Adjust the soft limit of open files to hard limit.
+// If |limit| equals 0, then directly return true.
+// If hard limit is less than |limit| than return false.
+bool AdjustOpenFileSoftLimit(uint64_t limit);
```
Comment (SeanHai) [inline src/client/utils.h:28]: Can this limit be adjusted dynamically?

File: conf/client.conf [hunk 2]
```diff
@@ -152,6 +152,14 @@ global.metricDummyServerStartPort=9000
 # 是否关闭健康检查: true/关闭 false/不关闭
 global.turnOffHealthCheck=true
 
+# minimal open file limit
```
Comment (SeanHai) [inline conf/client.conf:155]: minimal limit? has maximum limit?

File: conf/client.conf [hunk 3]
```diff
@@ -152,6 +152,14 @@ global.metricDummyServerStartPort=9000
...
+# if set value to 0, then will skip check and set open file limit
+# NOTE: open file limit will affect how may sockets we can create,
+#       the number of sockets is related to the number of chunkserver and mds in the cluster,
+#       and during some exception handling processes, client will create additional sockets
+#       the SAFE value is 2 * (#chunkserver + #mds)
+global.minOpenFileLimit=65536
```
Comment (SeanHai) [inline conf/client.conf:161]: each client 65536?

File: src/client/utils.h (hunk 1 above)
Comment (wu-hanqing) [reply src/client/utils.h:28]: we can use [prlimit](https://man7.org/linux/man-pages/man1/prlimit.1.html) to change running process's soft and hard limits

File: conf/client.conf (hunk 2 above)
Comment (wu-hanqing) [reply conf/client.conf:155]: This minimal limit affects how many TCP sockets the client can create, if it's too low, client can't create TCP sockets which will result in an IO Error.

A safe value for this is twice the number of cluster disks

File: conf/client.conf (hunk 3 above)
Comment (wu-hanqing) [reply conf/client.conf]: Lowering it to 1024 and it's fine in most cases.

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

File: src/client/config_info.h [hunk 4]
```diff
@@ -274,6 +274,7 @@ struct IOOption {
 struct CommonConfigOpt {
     bool mdsRegisterToMDS = false;
     bool turnOffHealthCheck = false;
+    uint32_t minimalOpenFiles = 65536;
```
Comment (Cyber-SiKu) [inline src/client/config_info.h:277]: Put the description of the configuration item in a comment.

File: src/client/utils.h [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+#ifndef SRC_CLIENT_UTILS_H_
+#define SRC_CLIENT_UTILS_H_
+
+#include <cstdint>
+
+namespace curve {
+namespace client {
+
+// Adjust the soft limit of open files to hard limit.
+// If |limit| equals 0, then directly return true.
```
Comment (Cyber-SiKu) [inline src/client/utils.h:26]: limit is uint64_t

File: src/client/utils.h (hunk 1 above)
Comment (Cyber-SiKu) [reply src/client/utils.h:28]: AdjustOpenFileSoftLimitToHardLimit is better?

File: src/client/utils.h (hunk 5 above)
Comment (Cyber-SiKu) [reply src/client/utils.h:26]: It seems to me that this code means to set softlimit to hardlimit. But hardlimit has a lower limit called `limit``.

**Review by Cyber-SiKu — COMMENTED** [review]
that is all

**Review by Cyber-SiKu — COMMENTED** [review]
Do you think my understanding is correct?

**Comment by wu-hanqing** [comment]
cicheck

File: src/client/config_info.h (hunk 4 above)
Comment (wu-hanqing) [reply src/client/config_info.h:277]: fixed

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

**Comment by wu-hanqing** [comment]
cicheck

File: src/client/utils.h (hunk 5 above)
Comment (wu-hanqing) [reply src/client/utils.h:26]: that's right.

**Review by Cyber-SiKu — APPROVED** [review]

**Review by SeanHai — APPROVED** [review]

**Pull request merged**